### PR TITLE
Kris prompt engineering

### PR DESCRIPTION
**Pull request by krisyifanyang**
This is prompt engineering branch based on RR-Watson Data branch

### CODE REVIEW

File: content/Watsonx/WatsonxAI/101.md [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+
+To receive a higher-quality response, provide an example of the kind of response you want. In technical terms, this is called Single Shot Prompting.
+
+![](./images/1.3.png)
+
+As you can see, providing one example before generating with your LLM is called Single Shot Prompting, but adding more examples into your prompt is common practice as well. Generally, increasing the number of examples is referred to as “Few Shot Prompting” and is a powerful tool to ensure you have a specific output.
+
+Copy paste to get started experimenting:
+
+```
```
Comment (maurelya) [inline content/Watsonx/WatsonxAI/101.md:46]: add the word `text` after ``` to fix the formatting I see on the webpage: 
Screenshot 2023-09-15 at 2 00 16 PM

File: content/Watsonx/WatsonxAI/101.md [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+The first change we can make is what model (LLM) we use to evaluate our prompt. This is one of the biggest changes you can make, as certain models are better built for specific tasks. The exercises later on this lab will force you to change the model you use if you want to answer some of the more challenging questions.
+
+In general, some models perform better working with summarization, key words, and semantics, while other models do better with structured text such as HTML, markdown, or JSON. The best way to figure out which models apply for your use case is to simply test them, but it is important to know that choice of model can make a big difference!
+
+watsonx.ai also provides multiple parameters for configuring how LLMs respond to a prompt. Selecting the correct parameters can often be more art than science. Investing time into understanding then tweaking these parameters will be rewarded by better responses.
+
+Explore these parameters with the following text:
+
+```
```
Comment (maurelya) [inline content/Watsonx/WatsonxAI/101.md:78]: add the word text after ``` to fix the formatting

File: content/Watsonx/WatsonxAI/101.md [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+LLMs have great potential, but they have no logic, no knowledge, and no domain expertise. Some use cases are a better fit than others: LLMs excel at tasks that involve generating generic text or common code patterns and transforming given input.
+
+If your prompt includes all the tips and best practices discussed here, yet you're not getting good results from any of the models, consider whether your use case might be one that LLMs just cannot handle well.
+
+For example, although we can get decent results for simple arithmetic, LLMs cannot generally do math well: [Researchers find that large language models struggle with math](https://venturebeat.com/business/researchers-find-that-large-language-models-struggle-with-math/)
+
+### 4.0 Balancing intelligence and security
+
+With great artificial intelligence comes higher security risks. Solutions like ChatGPT are Very Large language Models (VLLMs) with 175 billion parameters. They fine-tuned by the OpenAI team using an additional non-public Chat datasets plus a Reinforcement Learning Human Feedback (RLHF) dataset. ChatGPT is a chatbot-enabled LLM.
```
Comment (maurelya) [inline content/Watsonx/WatsonxAI/101.md:181]: This is well-written.

File: content/Watsonx/WatsonxAI/101.md [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+<table>
+<tr>
+<td>
+Write three sentences about donkeys
+</td>
+</tr>
+</table>
+  
+**Samples**
+```
```
Comment (maurelya) [inline content/Watsonx/WatsonxAI/101.md:210]: For the following exercises add the word `text` after ``` to fix the formatting

File: content/Watsonx/WatsonxAI/101.md [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+- As soon as the two puppies met at the park, their tails began to wag and they bounded around each other with glee.
+```
+```
+3 sentences about kittens:
+- The little kitten lapped up the milk with her tiny pink tongue, making a cute slurping sound.
+- The kitten nibbled on the treat, savoring every morsel of the delicious flavor.
+- As soon as the package was opened, the little kitten's eyes lit up with excitement.
+```
+
+See: [Example answer](109.md#1-generate)
```
Comment (maurelya) [inline content/Watsonx/WatsonxAI/101.md:223]: I am having issues going to the example answer page.
I am able to render the page if I include the full path `/watsonx/watsonxai/109#1-generate` like this

File: content/Watsonx/WatsonxAI/101.md [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+```
+
+```
+## Hall of shame: when LLMs go wrong
+Even the creators of LLMs cannot always fully anticipate or explain these models' output:
+[ChatGPT's creators can’t figure out why it won’t talk about Trump](https://www.semafor.com/
+article/02/03/2023/how-chatgpt-inadvertently-learned-to-avoid-talking-about-trump)
+```
+
+See: [Example answer](109.md#2-rewrite)
```
Comment (maurelya) [inline content/Watsonx/WatsonxAI/101.md:269]: `/watsonx/watsonxai/109#2-rewrite`

File: content/Watsonx/WatsonxAI/101.md [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+
+```
+As soon as the two dogs met at the park, their tails began to wag and they bounded around
+each other with glee. Their owners struck up a conversation, and soon found that they had
+much in common, bonding over their shared love of dogs and the outdoors. By the end of the
+day, new friendships had been formed, and both the dogs and their owners left the park with
+happy hearts and wagging tails.
+```
+
+See: [Example answer](109.md#3-summarize)
```
Comment (maurelya) [inline content/Watsonx/WatsonxAI/101.md:314]: `/watsonx/watsonxai/109#3-summarize`

File: content/Watsonx/WatsonxAI/101.md [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+
+```
+As soon as the package was opened, the little cat's eyes lit up with excitement.
+```
+
+```
+She pounced on the new toy, batting it around the room with joyous abandon.
+```
+
+See: [Example answer](109.md#4-text-extraction)
```
Comment (maurelya) [inline content/Watsonx/WatsonxAI/101.md:350]: `/watsonx/watsonxai/109#4-text-extraction`

File: content/Watsonx/WatsonxAI/101.md [hunk 9]
```diff
@@ -0,0 +1,352 @@
+# Lab 101: Prompt engineering
```
[the remark this answers is not in the document]
Comment (maurelya) [reply content/Watsonx/WatsonxAI/101.md:1]: When you are done editing the md files written in English, create new md files written in Spanish.
You can use Jake's [script](https://github.com/jbukuts/markdown-translate) to translate the md files

**Review by maurelya — CHANGES_REQUESTED** [review]

**Comment by ridgill** [comment]
Looks like this branch also contains all the changes from Rakesh's data lab.  Need to remove those commits before merging.  This branch should contain only the watsonx ai prompt lab content.

**Comment by krisyifanyang** [comment]
> Looks like this branch also contains all the changes from Rakesh's data lab. Need to remove those commits before merging. This branch should contain only the watsonx ai prompt lab content.

I will rebase based on the master

File: content/Watsonx/WatsonxAI/101.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (krisyifanyang) [reply content/Watsonx/WatsonxAI/101.md:1]: we should direct them to the watsonx main page then create a project then we can see the prompt lab, but fyi it is here dataplatform.cloud.ibm.com/wx

**Comment by ridgill** [comment]
newer PR opened

**Pull request closed without merging**